### ISSUE TITLE
Update prometheus response in /metrics

### ISSUE THREAD

**Issue by bartekn** [stellar/go#2599]
From @jacekn:
> We expose _bucket metrics, which in prometheus are histograms, but they are actually summaries that use the quantile labels. _buckets metrics should be counters with le labels.

We probably also return data in prometheus format by default (with an optional JSON response when passing a GET param).

Assignees should have a short chat with @jacekn who will explain what can be improved to make sure Prometheus can consume data without any issues.

**Comment by jacekn** [comment]
It would be good to also improve `horizon_requests_*` metrics 
Currently we have:
`horizon_requests_total_count`
`horizon_requests_succeeded_count`
`horizon_requests_failed_count`

There is no need to expose all 3. `total` and `failed` is enough, we can calculate `successful` if needed.

Also we're currently seeing about 30% of `failed` requests. I think it may be because 4xx are counted as failed. It would be good to differentiate between client errors and servers error for alerting purposes. It's worth knowing how many 4xx we're serving but 5xx are likely something to alert on.

Alternative approach to having 3 metrics is one metric with HTTP status code exposed as label, for example:
```
horizon_requests_total{status="200"} 123
horizon_requests_total{status="301"} 32
horizon_requests_total{status="404"} 43
horizon_requests_total{status="501"} 2
```

**Comment by bartekn** [comment]
@jacekn I'm going to start fixing these issues soon. I'd like to recreate Horizon Kibana dashboard in Prometheus and I started wondering how to do it. For HTTP requests/responses I'd like to have a detailed breakdown of count (possibly with 1m, 5m, 15m rate) and duration (possibly with percentiles):
* per response code,
* per route (ex. `/accounts/{id}/payments`
* streaming/non-streaming.

So I wonder how to expose metrics to make it as easy as possible to use in Prometheus. Is it OK to add multiple labels to metric like:
```
# total number of 200 responses for non-streaming requests to all /accounts/{id}/payments
horizon_requests_total{status="200", route="/accounts/{id}/payments", streaming="false"} 123
```
 We can also have a quick chat if it's faster for you.

**Comment by bartekn** [comment]
@jacekn the issues here should be solved by https://github.com/stellar/go/pull/2846. It contains example `/metrics` response. Can you check if it looks good?

**Comment by bartekn** [comment]
Closed in #2846.
